service-api/module/Application/src/DWP/DwpApi/DwpApiService.php

```diff
@@ -0,0 +1,243 @@
+        }
+    }
+
+    public function compareRecords(
+        CaseData $caseData,
+        DetailsResponseDTO $detailsResponseDTO,
+        CitizenResponseDTO $citizenResponseDTO,
+        string $nino
+    ): bool {
+        if ($caseData->idMethodIncludingNation) {
+            /** @psalm-suppress PossiblyNullArgument */
+            $submittedNino = strtoupper(
+                preg_replace(
+                    '/(\s+)|(-)/',
+                    '',
+                    $nino
+                )
+            );
+
+            /** @psalm-suppress PossiblyNullArgument */
+            $returnedNino = strtoupper(
+                preg_replace(
+                    '/(\s+)|(-)/',
+                    '',
+                    $detailsResponseDTO->nino()
+                )
+            );
+
+            if ($citizenResponseDTO->type() !== 'MatchResult') {
+                $this->logger->info("Match Result missing from response.");
+                return false;
+            }
+
+            if ($submittedNino !== $returnedNino) {
+                $this->logger->info("Submitted NINO does not match NINO in details response.");
+                return false;
+            }
+
+            return true;
+        } else {
+            throw new DwpApiException('National Insurance Number not set.');
+        }
+    }
+
+    /**
+     * @throws GuzzleException
+     * @throws DwpApiException
+     */
+    public function makeCitizenMatchRequest(
+        CitizenRequestDTO $citizenRequestDTO,
+    ): CitizenResponseDTO {
+        $this->authCount++;
+        $responseArray = [];
+        try {
+            $postBody = $this->constructCitizenRequestBody($citizenRequestDTO);
+
+            $response = $this->guzzleClient->request(
+                'POST',
+                $this->matchPath,
+                [
+                    'headers' => $this->makeHeaders(),
+                    'json' => $postBody
+                ]
+            );
+            $responseArray = json_decode($response->getBody()->getContents(), true);
+        } catch (ClientException $clientException) {
+            if (
+                $clientException->getResponse()->getStatusCode() == Response::STATUS_CODE_401 &&
+                $this->authCount < 2
+            ) {
+                $this->authApiService->authenticate();
+                $this->makeCitizenMatchRequest($citizenRequestDTO);
+            } else {
+                $response = $clientException->getResponse();
+                $responseBodyAsString = $response->getBody()->getContents();
+                $this->logger->error('GuzzleDwpApiException: ' . $responseBodyAsString);
+                throw $clientException;
+            }
+        } catch (\Exception $exception) {
+            throw new DwpApiException($exception->getMessage());
+        }
+        return new CitizenResponseDTO(
+            $responseArray
+        );
+    }
+
+    public function constructCitizenRequestBody(
```

Comment:
CitizenRequestDTO exists to translate our local data to something acceptable for the citizen request endpoint, so I suspect this function should be inside the DTO class